hazelcast/src/main/java/com/hazelcast/datalink/impl/DataLinkServiceImpl.java

```diff
@@ -93,18 +92,19 @@ private void put(DataLinkConfig config, DataLinkSource source, boolean replace)
                                     + "', ignoring it: " + e, e);
                         }
                     }
-                    return pair(createDataLinkInstance(config), source);
+                    return new DataLinkEntry(createDataLinkInstance(config), config, source);
                 });
     }
 
     @Override
-    public void createSqlDataLink(String name, String type, Map<String, String> options, boolean replace) {
-        put(toConfig(name, type, options), SQL, replace);
+    public void replaceSqlDataLink(String name, String type, Map<String, String> options) {
+        put(toConfig(name, type, options), SQL);
     }
 
     @Override
-    public boolean existsDataLink(String name) {
-        return dataLinks.containsKey(name);
+    public boolean existsConfigDataLink(String name) {
```

Comment:
Why do we check only `CONFIG`-sourced data links? `existsDataLink()` is useful for multiple scenarios in my consequent PRs, for example.

Reply:
You can re-add it. Now I don't need it. The existence of a sql-sourced datalink is solved by `putIfAbsent` operation. This method handles cases when we want to avoid overwriting a config-sourced datalink.